### PR TITLE
Upgrade to Elasticsearch 7.0.0

### PR DESCRIPTION
**Pull request by duxet**
#179

### CODE REVIEW

**Comment by lukas-vlcek** [comment]
@duxet I am trying your PR but compilation fails for me. Here is my log:

```
$ git fetch origin pull/180/head:r7.0.0
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 5 (delta 0), reused 1 (delta 0), pack-reused 0
Unpacking objects: 100% (5/5), done.
From https://github.com/vvanholl/elasticsearch-prometheus-exporter
 * [new ref]         refs/pull/180/head -> r7.0.0
 * [new tag]         5.6.16.0           -> 5.6.16.0
 * [new tag]         6.7.1.0            -> 6.7.1.0

$ git checkout r7.0.0
Switched to branch 'r7.0.0'

$ git log --oneline -n 5
558cbd7 (HEAD -> r7.0.0) Upgrade to Elasticsearch 7.0.0
d52c619 (origin/master, origin/HEAD, master) Prepare next release
e7c69ed (tag: 6.7.1.0) Release version 6.7.1.0
76c0aab Merge pull request #177 from vvanholl/es671
47a1e6f Temporary use compuscene/gradle:5.3.1-jdk12 waiting official Gradle image

$ java --version
openjdk 12 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)

$ gradle --version

------------------------------------------------------------
Gradle 5.3.1
------------------------------------------------------------

Build time:   2019-03-28 09:09:23 UTC
Revision:     f2fae6ba563cfb772c8bc35d31e43c59a5b620c3

Kotlin:       1.3.21
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          12 (Oracle Corporation 12+33)
OS:           Mac OS X 10.14.3 x86_64

$ gradle clean check
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :
=======================================
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 5.3.1
  OS Info               : Mac OS X 10.14.3 (x86_64)
  JDK Version           : 12 (Oracle Corporation 12 [OpenJDK 64-Bit Server VM 12+33])
  JAVA_HOME             : /Library/Java/JavaVirtualMachines/openjdk-12.jdk/Contents/Home
  Random Testing Seed   : AC407754182B6F15
=======================================
Host: MacBook-Pro/10.43.21.28
Gradle: 5.3.1 JVM: 12 (Oracle Corporation 12+33) Groovy: 2.5.4
Build: group: 'org.elasticsearch.plugin.prometheus', name: 'prometheus-exporter', version: '7.0.0.0-SNAPSHOT'
Timestamp: 2019-04-12T09:29:56.235583+02:00[Europe/Prague]

> Task :compileJava
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java:25: error: wrong number of type arguments; required 1
public class NodePrometheusMetricsAction extends Action<NodePrometheusMetricsRequest, NodePrometheusMetricsResponse,
                                                       ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java:49: error: TransportNodePrometheusMetricsAction is not abstract and does not override abstract method doExecute(Task,NodePrometheusMetricsRequest,ActionListener<NodePrometheusMetricsResponse>) in TransportAction
public class TransportNodePrometheusMetricsAction extends HandledTransportAction<NodePrometheusMetricsRequest,
       ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java:61: error: no suitable constructor found for HandledTransportAction(Settings,String,ThreadPool,TransportService,ActionFilters,IndexNameExpressionResolver,NodePromet[...]::new)
        super(settings, NodePrometheusMetricsAction.NAME, threadPool, transportService, actionFilters,
        ^
    constructor HandledTransportAction.HandledTransportAction(String,TransportService,ActionFilters,Supplier<NodePrometheusMetricsRequest>) is not applicable
      (actual and formal argument lists differ in length)
    constructor HandledTransportAction.HandledTransportAction(String,TransportService,ActionFilters,Reader<NodePrometheusMetricsRequest>) is not applicable
      (actual and formal argument lists differ in length)
    constructor HandledTransportAction.HandledTransportAction(String,TransportService,ActionFilters,Reader<NodePrometheusMetricsRequest>,String) is not applicable
      (actual and formal argument lists differ in length)
    constructor HandledTransportAction.HandledTransportAction(String,boolean,TransportService,ActionFilters,Supplier<NodePrometheusMetricsRequest>) is not applicable
      (actual and formal argument lists differ in length)
    constructor HandledTransportAction.HandledTransportAction(String,boolean,TransportService,ActionFilters,Reader<NodePrometheusMetricsRequest>) is not applicable
      (actual and formal argument lists differ in length)
    constructor HandledTransportAction.HandledTransportAction(String,boolean,TransportService,ActionFilters,Reader<NodePrometheusMetricsRequest>,String) is not applicable
      (actual and formal argument lists differ in length)
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java:69: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java:204: warning: [deprecation] logger in TransportAction has been deprecated
            if (logger.isTraceEnabled()) {
                ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java:205: warning: [deprecation] logger in TransportAction has been deprecated
                logger.trace("Return response: [{}]", response);
                ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/plugin/prometheus/PrometheusExporterPlugin.java:56: error: cannot infer type arguments for ActionHandler<>
                new ActionHandler<>(NodePrometheusMetricsAction.INSTANCE, TransportNodePrometheusMetricsAction.class)
                ^
  reason: cannot infer type-variable(s) Request,Response
    (argument mismatch; NodePrometheusMetricsAction cannot be converted to Action<Response>)
  where Request,Response are type-variables:
    Request extends ActionRequest declared in class ActionHandler
    Response extends ActionResponse declared in class ActionHandler
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/NodePrometheusRequestBuilder.java:29: error: incompatible types: NodePrometheusMetricsAction cannot be converted to Action<NodePrometheusMetricsResponse>
        super(client, action, new NodePrometheusMetricsRequest().local(true));
                      ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java:34: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java:40: error: method does not override or implement a method from a supertype
    @Override
    ^
/Users/lvlcek/projects/elasticsearch-prometheus-exporter/src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java:45: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
9 errors
2 warnings

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 21s
2 actionable tasks: 2 executed
```
Are you able to run tests via `gradle check`?

**Comment by duxet** [comment]
I have fixed all build errors and warnings, however i haven't checked yet manually if the plugin works correctly.

**Comment by lukas-vlcek** [comment]
There is a set of integration tests that are run when `gradle check` is run. It starts ES cluster with [2 nodes](https://github.com/vvanholl/elasticsearch-prometheus-exporter/blob/d52c6194e46bc0debb9efd3083f58ef724287882/build.gradle#L106), deploy Prometheus plugin to all the nodes and them performs all the tests found under [src/test/resources/rest-api-spec/test/resthandler](https://github.com/vvanholl/elasticsearch-prometheus-exporter/tree/master/src/test/resources/rest-api-spec/test/resthandler) folder (quite a bit of tests). If these tests are passing there is a good change the plugin works as expected (these are real integration tests). You should see some confirmation that all the tests passed.

I will do code review during the weekend. Thanks for the PR!

File: src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java [hunk 1]
```diff
@@ -37,11 +36,6 @@ public NodePrometheusMetricsResponse newResponse() {
         throw new UnsupportedOperationException("usage of Streamable is to be replaced by Writeable");
     }
 
-    @Override
-    public NodePrometheusRequestBuilder newRequestBuilder(ElasticsearchClient client) {
```
Comment (lukas-vlcek) [inline src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java]: Can you please remove `ElasticsearchClient` reference from imports now? It is not longer needed in this file.

File: src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java [hunk 2]
```diff
@@ -52,22 +55,22 @@
     private final Settings settings;
     private final ClusterSettings clusterSettings;
     private final PrometheusSettings prometheusSettings;
+    private final Logger logger = LogManager.getLogger(getClass());
 
     @Inject
-    public TransportNodePrometheusMetricsAction(Settings settings, ThreadPool threadPool, Client client,
```
Comment (lukas-vlcek) [inline src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java]: Remove relevant `import org.elasticsearch.threadpool.ThreadPool` please.

File: src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java [hunk 3]
```diff
@@ -52,22 +55,22 @@
...
+    public TransportNodePrometheusMetricsAction(Settings settings, Client client,
                                                 TransportService transportService, ActionFilters actionFilters,
-                                                IndexNameExpressionResolver indexNameExpressionResolver,
```
Comment (lukas-vlcek) [inline src/main/java/org/elasticsearch/action/TransportNodePrometheusMetricsAction.java]: Remove relevant `import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver` please.

**Comment by duxet** [comment]
Done :)

**Comment by lukas-vlcek** [comment]
@vvanholl I am handing it over to you :-)

**Comment by davidkarlsen** [comment]
@vvanholl ping?

**Comment by vvanholl** [comment]
Hi @lukas-vlcek @davidkarlsen I am working on it today

**Comment by vvanholl** [comment]
Thank you @duxet and @lukas-vlcek for supporting ES 7
It's time now to release !

**Pull request merged**